Exclude "gtest" from installing along with the library

cartoonist opened the pull request:
This avoids clashing with the installed version of "googletest". I want to create a conda-forge recipe for this fork (`sdsl-lite-vgteam`) and this might create a conflict with "gtest" package.

cartoonist committed:
Exclude "gtest" from installing along with the library

This voids clashing with the installed googletest.

jltsiren commented:
It would be better if you kept an option to install googletest along with SDSL. Some of our other projects (such as GBWT) rely on the version of googletest that comes with SDSL. I don't know if their tests would work with versions of googletest installed from other sources, and I don't have the resources to find out.

cartoonist committed:
Add an option for installing gtest alongside

This commit add `WITH_GOOGLETEST` cmake option. When is ON gtest will be
installed alongside the library. To void breaking other projects relying
on installed version of googletest, this option is ON by default.

cartoonist commented:
If it breaks other projects like GBWT, I have found a workaround for the conflict in conda (i.e. patching the CMake file internally using conda before building). However, I think your suggestion for making it optional is a good idea, since shipping dependencies along with the library might not be generally ideal as it replaces already installed ones. So, I added CMake option `WITH_GOOGLETEST` to control the installation and it is ON by default for backward compatibility.

cartoonist commented:
Sorry I am writing here, the Issue tracker is disabled. I created a conda recipe on conda-forge for this fork, but they are kind of reluctant to publish it as `vgteam-sdsl-lite` as it conflicts with the original package. They offer either merged with the upstream or rebrand the library (see [here](https://github.com/conda-forge/staged-recipes/pull/20669) about the recipe itself and [here](https://github.com/conda-forge/conda-forge.github.io/issues/1851) is a discussion about publishing a fork of a package). Nevertheless, they can publish it as `vgteam-sdsl-lite` if neither options work.

I'd also like to ask about the status of this fork of sdsl-lite. As far as I know, the original repository does not accept PRs anymore. There are two active forks that I know, one is the vgteam's and the other is maintained by SeqAn team I think (branded as sdslv3). Do you know if there is any plan for the future of the library? e.g. taking over the upstream or agreeing on a single fork to accumulate all efforts?

jltsiren commented:
First, I'm on strike, like other researchers at the University of California. While it's possible that the strike will end in mid-December, I don't expect that anything will get done this year.

The problem with all SDSL forks is that nobody has the developer resources to maintain a fully supported version of the library. As far as I understand, SDSL 3 is effectively a single person doing the changes needed for SeqAn. This fork is me doing the changes needed for the vg ecosystem. The forks have diverged significantly, and merging them would require nontrivial effort.
